mlflow/pipelines/steps/split.py

```diff
@@ -61,10 +61,34 @@ def _get_split_df(input_df, hash_buckets, split_ratios):
     return train_df, validation_df, test_df
 
 
+def _parallelize(data, func, num_of_processes=8):
+    import numpy as np
+    import pandas as pd
+    from multiprocessing import Pool
+
+    data_split = np.array_split(data, num_of_processes)
```

Comment:
Ideally the num_of_processes should be data dependent. e.g. I don't see a reason to split it if it's only 10 MB or smaller. The actual optimal value is probably a piece-wise linear function to data size. 

Not sure where to make the cut. It can probably be determined experimentally by varying the data size vs num_of_processes, if you'd like to give it a try:) Otherwise, apply some simply binary logic, if smaller than 10 MB, use 1 process, otherwise, use the max pool size.

Reply:
@jinzhang21 agreed, there's probably some heuristic we can follow for the optimal `n`.

my thought process with this design is for small datasets the total duration of the split step is already low, so introducing some additional latency will likely not be noticeable. but for large data, this will provide significant speedup.